servatrice/src/servatrice_database_interface.cpp

```diff
@@ -1128,3 +1130,91 @@ int Servatrice_DatabaseInterface::checkNumberOfUserAccounts(const QString &email
 
     return 0;
 }
+
+bool Servatrice_DatabaseInterface::addForgotPassword(const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	if (!updateUserToken(PasswordHasher::generateActivationToken(), user))
+		return false;
+
+	QSqlQuery *query = prepareQuery("insert into {prefix}_forgot_password (name,requestDate) values (:username,NOW())");
+	query->bindValue(":username", user);
+	if (execSqlQuery(query))
+		return true;
+
+	return false;
+}
+
+bool Servatrice_DatabaseInterface::removeForgotPassword(const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	QSqlQuery *query = prepareQuery("delete from {prefix}_forgot_password where name = :username");
+	query->bindValue(":username", user);
+	if (execSqlQuery(query))
+		return true;
+
+	return false;
+}
+
+bool Servatrice_DatabaseInterface::doesForgotPasswordExist(const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	QSqlQuery *query = prepareQuery("select count(name) from {prefix}_forgot_password where name = :user_name AND requestDate > (now() - interval :minutes minute)");
+	query->bindValue(":user_name", user);
+	query->bindValue(":minutes", QString::number(server->getForgotPasswordTokenLife()));
+
+	if (!execSqlQuery(query))
+		return false;
+
+	if (query->next())
+		if (query->value("count(name)").toInt() > 0)
+			return true;
+
+	return false;
+}
+
+bool Servatrice_DatabaseInterface::updateUserToken(const QString & token, const QString &user)
+{
+	if (!checkSql())
+		return false;
+
+	if (token.isEmpty() || user.isEmpty())
+		return false;
+
+	QSqlQuery *query = prepareQuery("update {prefix}_users set token = :token where name = :user_name");
```

Comment:
Can you tell me how this update method is used? I'm a little leery of reusing the users token field - I thought that was used only for non-activated accounts? It seems like we should store the token in the forgot password table, and then only check there when we get the reset-with-token proto message.